pkg/action/custom_resource_lister.go

```diff
@@ -0,0 +1,170 @@
+	}
+
+	err := o.config.Client.Get(ctx, csvKey, &csv)
+	if err != nil {
+		return nil, fmt.Errorf("could not get %s CSV on cluster: %s", csvKey.String(), err)
+	}
+
+	// check if owned CRDs are defined on the csv
+	if len(csv.Spec.CustomResourceDefinitions.Owned) == 0 {
+		return nil, fmt.Errorf("no owned CustomResourceDefinitions specified on CSV %s, no custom resources to display", csvKey.String())
+	}
+
+	// check CSV is not in a failed state (to ensure some OLM multitenancy rules are not violated)
+	if csv.Status.Phase != v1alpha1.CSVPhaseSucceeded {
+		return nil, fmt.Errorf("CSV underlying operator is not in a succeeded state")
+	}
+
+	return csv.Spec.CustomResourceDefinitions.Owned, nil
+}
+
+// List takes in a CRD description and finds the associated CRs on-cluster.
+// List can return a potentially unbounded list that callers may need to paginate.
+func (o *OperatorListOperands) list(ctx context.Context, crdDesc v1alpha1.CRDDescription, namespaces []string) (*unstructured.UnstructuredList, error) {
+	result := &unstructured.UnstructuredList{}
+
+	list := unstructured.UnstructuredList{}
+	gvk := schema.GroupVersionKind{
+		Group:   crdDesc.Name,
+		Version: crdDesc.Version,
+		Kind:    crdDesc.Kind,
+	}
+	result.SetGroupVersionKind(gvk)
+	if err := o.config.Client.List(ctx, &list); err != nil {
+		return nil, err
+	}
+
+	// trim down CRs in list to match target namespaces
+	if namespaces == nil {
+		return &list, nil
+	}
+	for _, cr := range list.Items {
+		if cr.GetNamespace() == "" || inNamespace(cr.GetNamespace(), namespaces) {
+			result.Items = append(result.Items, cr)
+		}
+	}
+
+	return result, nil
+}
+
+// ListAll wraps the above functions to provide a convenient command to go from package/namespace to custom resources.
+func (o *OperatorListOperands) listAll(ctx context.Context, opKey types.NamespacedName) (*unstructured.UnstructuredList, error) {
+	operator, err := o.findOperator(ctx, opKey)
+	if err != nil {
+		return nil, err
+	}
+
+	crdDescs, err := o.unzip(ctx, operator)
+	if err != nil {
+		return nil, err
+	}
+
+	// find all namespaces associated with operator via operatorgroup
+	// query for CRs in these namespaces
+	ogList := v1.OperatorGroupList{}
+	options := client.ListOptions{Namespace: o.config.Namespace}
+	if err := o.config.Client.List(ctx, &ogList, &options); err != nil {
+		return nil, err
+	}
+	if len(ogList.Items) != 1 {
+		return nil, fmt.Errorf("unexpected number (%d) of operator groups found in namespace %s", len(ogList.Items), o.config.Namespace)
+	}
+	namespaces := ogList.Items[0].Spec.TargetNamespaces
+
+	var result unstructured.UnstructuredList
+	for _, crd := range crdDescs {
+		list, err := o.list(ctx, crd, namespaces)
+		if err != nil {
+			return nil, err
+		}
+		for _, cr := range list.Items {
+			result.Items = append(result.Items, cr)
+		}
+	}
+	return &result, nil
+}
+
+func inNamespace(ns string, namespaces []string) bool {
+	for _, n := range namespaces {
+		if n == ns {
```

Comment:
When watching all namespaces, a targetNamespace will be the empty string.
```suggestion
		if n == ns || n == "" {
```

Reply:
good catch